File: updated_code/code/java/1880-check-if-word-equals-summation-of-two-words.java
package com.check;

import java.util.regex.Pattern;

public class Solution {
    public boolean isSumEqual(String firstWord, String secondWord, String targetWord) {
        // Convert each word to numerical value
        int firstValue = numValue(firstWord);
        int secondValue = numValue(secondWord);
        int targetValue = numValue(targetWord);

        // Check if the sum of the numerical values of firstWord and secondWord equals the numerical value of targetWord
        return firstValue + secondValue == targetValue;
    }

    private int numValue(String word) {
        int value = 0;
        for (int i = 0; i < word.length(); i++) {
            value = value * 10 + (word.charAt(i) - 'a');
        }
        return value;
    }
}

File: updated_code/code/python/1880-check-if-word-equals-summation-of-two-words.py
class Solution:
    def is_sum_equal(self, first_word: str, second_word: str, target_word: str) -> bool:
        """
        This method checks if the sum of the numerical values of two words equals the numerical value of a target word.

        Args:
            first_word (str): The first word.
            second_word (str): The second word.
            target_word (str): The target word.

        Returns:
            bool: True if the sum of the numerical values of the two words equals the numerical value of the target word, False otherwise.
        """
        def get_numerical_value(word: str) -> int:
            """
            This method calculates the numerical value of a word.

            Args:
                word (str): The word.

            Returns:
                int: The numerical value of the word.
            """
            return int(''.join(str(ord(c) - ord('a')) for c in word))

        return get_numerical_value(first_word) + get_numerical_value(second_word) == get_numerical_value(target_word)